### PR TITLE
Overhaul endpoints and data mapping

### PR DESCRIPTION
**Pull request by tommyjackson**
Spent some time working with Tero on how to improve our data fetching and he helped me come up with some great solutions, as well as added some new fields to make some things easier for us. With this PR, the BFF layer has 4 endpoints to access Tero's db, and exposes 3 endpoints for the app to access.

This is how data fetching will work now (note that a `Season` object is an individual numbered season within a specified league, so there are multiple seasons with `name = "2"` for example, but they would be under different `League` parent objects):
1. On app launch, fetch all leagues with seasons `/leagues?populate=seasons`. Tero added the number of weeks to a `Season` object, so now based on this one call we can supply all of our league/season/week configuration options
2. Next, once the app determines the selected league and season (either via persistence or defaults), we load all of the teams and matches for that league's selected season. This is done by combining 2 calls to Tero's db.
- `/seasons/$seasonId?populate=matches.games`: this gives us all of the matches, with games included, only for the selected league and season configuration
- `"/teams?_id={id1}&_id={id2}"...`: The original leagues call we made returns a list of team ids associated with each returned season, so passing those along into this call gets us only the teams for the selected league and season configuration.
- With those 2 responses, we can then go through and calculate the wins/losses for each team and return all of the matches and teams (with w/l data appended) in a single response since its smaller because its only for one league/season. I did the combining logic in the `Controller` since to me it made more sense to have our `Repository`s be just a wrapper for calls we need to access Tero's db, and have the `Controller` be in charge of chaining any logic for calls that are app makes.
3. With the above 2 calls, we basically get everything we need to configure the schedule and standings stuff.
4. When navigating to a team details page, we can get the players associated with the team for that season by using `/players?on_teams_date={date}&on_teams_team_ids={team_id}`. Tero went and added `start_date` and `end_date` fields to each season, so using the season's end date with that call gives us the roster to associate with that team that makes the most logical sense, given roster moves and subs throughout a season. 

This is quite a big overhaul and i ended up deleting all of the stuff that wasn't being used at this point. i also removed the distinction between local/remote data models bc it didn't seem as necessary anymore (or maybe it never was and i'm just dumb).

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -86,6 +86,7 @@ dependencies {
     implementation("io.ktor:ktor-server-resources")
     implementation("io.ktor:ktor-server-default-headers")
     implementation("io.ktor:ktor-server-html-builder")
+    implementation("io.ktor:ktor-server-content-negotiation")
```
Comment (j-roskopf) [inline build.gradle.kts:89]: we already declare this

File: src/main/kotlin/com/boostyboys/mcs/Application.kt [hunk 2]
```diff
@@ -37,10 +40,21 @@ private const val LOCAL_PORT = 8080
 /**
  * An entry point of the embedded-server
  **/
+@OptIn(ExperimentalSerializationApi::class)
 fun main() {
     val port = System.getenv("PORT")?.toInt() ?: LOCAL_PORT
     embeddedServer(Netty, port = port) {
         kodeinApplication {
+            install(ContentNegotiation) {
```
Comment (j-roskopf) [inline src/main/kotlin/com/boostyboys/mcs/Application.kt:48]: we already have this installed in the bindHttpClient method - is this doing something different?

**Review by j-roskopf — APPROVED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (tommyjackson) [reply build.gradle.kts:89]: we had the client one, but we needed the server one too when i needed to deserialize a request body and not just query params. spent quite some time trying to figure out what wasn't working bc i thought the same that we already had what we needed

File: src/main/kotlin/com/boostyboys/mcs/Application.kt (hunk 2 above)
Comment (tommyjackson) [reply src/main/kotlin/com/boostyboys/mcs/Application.kt:48]: same as the other comment

**Pull request merged**